circle.yml

```diff
@@ -36,3 +36,28 @@ jobs:
             if [ "${CIRCLE_BRANCH}" == "release" ]; then
               make release
             fi
+  publish:
+    machine:
+      image: ubuntu-1604:202007-01
+    working_directory: /home/circleci/logspout
+    environment:
+      DEBUG: true
+    steps:
+      - checkout
+      - run: mkdir -vp ~/.docker/cli-plugins/
+      - run: curl --silent -L --output ~/.docker/cli-plugins/docker-buildx https://github.com/docker/buildx/releases/download/v0.3.1/buildx-v0.3.1.linux-amd64
+      - run: chmod a+x ~/.docker/cli-plugins/docker-buildx
+      - run: docker login -u "$DOCKER_USERNAME" -p "$DOCKER_PASSWORD"
+      - run: docker run -it --rm --privileged tonistiigi/binfmt --install all
+      - run: docker buildx create --use --name mybuilder
+      - run: docker buildx build --push --platform linux/arm64,linux/amd64 -t ${DOCKER_USERNAME}/logspout:${CIRCLE_BRANCH} .
+      - run: |
+            if [ "$CIRCLE_BRANCH" = "master" ]; then
+               docker buildx build --push --platform linux/arm64,linux/amd64 -t ${DOCKER_USERNAME}/logspout:latest .
```

Comment:
i'm not familiar with buildx. does this end up pushing two different images? if so, what are their names/tags?

Reply:
When it will run on master branch one image with two different tags will be published 

1) gliderlabs/logspout:master 

2) gliderlabs/logspout:latest 

When it will run on release branch, image will be publish in dockerhub as gliderlabs/logspout:v3.2.15

Reply:
That makes sense. However my question was about the two architectures. Do we wind up with one for `linux/arm64` and one for `linux/amd64`? As an end-user, how would I get the `linux/arm64` version for example?

Reply:
Looks like docker pulls the correct image for your target architecture automatically